src/NServiceBus.Testing/Test.cs

```diff
@@ -140,75 +50,34 @@ public static Saga<T> Saga<T>(Guid sagaId) where T : Saga, new()
+        /// Specify a test for a message handler of type T for a given message of type TMessage.
         /// </summary>
-        public static Handler<T> Handler<T>() where T : new()
+        public static Handler<THandler> Handler<THandler>() where THandler : new()
         {
-            var handler = (T)Activator.CreateInstance(typeof(T));
-
-            return Handler(handler);
+            return Handler(new THandler());
         }
 
         /// <summary>
-        ///     Specify a test for a message handler while supplying the instance to
-        ///     test - injects the bus into a public property (if it exists).
+        /// Specify a test for a message handler while supplying the instance to
+        /// test - injects the bus into a public property (if it exists).
         /// </summary>
-        public static Handler<T> Handler<T>(T handler)
+        public static Handler<THandler> Handler<THandler>(THandler handler)
         {
-            Func<IBus, T> handlerCreator = b => handler;
-            var prop = typeof(T).GetProperties().FirstOrDefault(p => p.PropertyType == typeof(IBus));
-            if (prop != null)
-            {
-                handlerCreator = b =>
-                {
-                    prop.SetValue(handler, b, null);
-                    return handler;
-                };
-            }
-
-            return Handler(handlerCreator);
+            return new Handler<THandler>(handler);
         }
 
         /// <summary>
-        ///     Specify a test for a message handler specifying a callback to create
-        ///     the handler and getting an instance of the bus passed in.
-        ///     Useful for handlers based on constructor injection.
+        /// Specify a test for a message handler specifying a callback to create
+        /// the handler and getting an instance of the bus passed in.
+        /// Useful for handlers based on constructor injection.
         /// </summary>
-        public static Handler<T> Handler<T>(Func<IBus, T> handlerCreationCallback)
+        public static Handler<T> Handler<T>(Func<IMessageSession, T> handlerCreationCallback)
         {
-            bus = new StubBus(messageCreator);
-
-            var handler = handlerCreationCallback.Invoke(bus);
-
-            var isHandler = (from i in handler.GetType().GetInterfaces()
-                             let args = i.GetGenericArguments()
-                             where args.Length == 1
-                             where typeof(IHandleMessages<>).MakeGenericType(args[0]).IsAssignableFrom(i)
-                             select i).Any();
-
-            if (!isHandler)
-            {
-                throw new ArgumentException("The handler object created does not implement IHandleMessages<T>.", "handlerCreationCallback");
-            }
-
-            return new Handler<T>(handler, bus);
+            throw new NotImplementedException();
```

Comment:
@timbussmann @hmemcpy It feels like this method is no longer required as we don't need to give them a bus to inject into the constructor. Are you happy obsolete it?